core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -360,6 +360,22 @@ public QueryAssert returnsEmptyResult()
             });
         }
 
+        public QueryAssert hasPlan(PlanMatchPattern expectedPlan)
```

Comment:
in it's current form it's not much different from `isNotFullyPushedDown`

- do we need a separate method? it would make sense if we did not calculate & compare the results at all (faster), but can we do this?
- should `isNotFullyPushedDown` delegate here?

Reply:
`isNotFullyPushedDown` can reuse this now.
We need separate method (which doesn't verify results) to make sure it's fast enough. Running join queries over tpch data generally means a test may run for > 1m.

Or I can remove this in favour of `isNotFullyPushedDown` if the test's runtime isn't a huge concern.

Reply:
I'm removing this - there's not a very large difference with or without verifying results.